src/backend/FrozenLakeEnv.js

```diff
@@ -0,0 +1,118 @@
+/* Based on the OpenAI Gym implementation
+ * https://github.com/openai/gym/blob/master/gym/envs/toy_text/frozen_lake.py
+ */
+
+export class FrozenLakeEnv {
+	/**
+	 * Env Class
+	 */
+
+
+
+	constructor(desc=null, map_name="4x4", is_slippery=true) {
+
+	  this.MAPS = {
+        "4x4": [
+            "SFFF",
+            "FHFH",
+            "FFFH",
+            "HFFG"
+        ],
+        "8x8": [
+            "SFFFFFFF",
+            "FFFFFFFF",
+            "FFFHFFFF",
+            "FFFFFHFF",
+            "FFFHFFFF",
+            "FHHFFFHF",
+            "FHFFHFHF",
+            "FFFHFFFG"
+        ],
+    };
+
+    if (desc === null && map_name === null) {
+      this.desc = this.MAPS["4x4"];
+    } else if (desc===null) {
+      this.desc = this.MAPS[map_name];
+    }
+
+    this.nrow = this.desc.length;
+    this.ncol = this.desc[0].length;
+    this.reward_range = (0, 1);
+
+    this.nA = 4;
+    this.nS = this.nrow * this.ncol;
+
+    this.reset();
+  }
+
+	// Returns a tensor describing the state space of the env
+	// Will be used to determine the input dimensionality of the model
+	getStateSpace() {
+    return 2;
```

Comment:
Shouldn't this return `this.nS`?

Reply:
That's a good point. The state is a vector [x, y], so I just returned its shape (should have been [1, 2], now that I think of it....). But I was really sleepy, that doesn't make sense. Maybe returning [4,4] (or [8,8]) would make more sense?

Reply:
It should return whatever the shape of the input to the model will be. If we're just gonna feed the 2D coords into the model then the shape can be 2 or [2] I think